File: Pathfinder.Web/ClientApp/src/models/Category.ts
import { ICategory } from './Interfaces/ICategory'
export class Category implements ICategory {
    id: number;
    name: string;
    description: string;
    constructor (id: number, name: string, description: string){
        this.id = id;
        this.name = name;
        this.description = description;
    }
}

File: Pathfinder.Core/Entities/Product/Category.cs
using Pathfinder.Core.Entities.Base;
using System.ComponentModel.DataAnnotations;

namespace Pathfinder.Core.Entities.Product
{
    public class Category
    {
        [Key]
        public CategoryType CategoryType { get; set; }
        [Required, StringLength(80)]
        public string Name { get; set; }
        public string Description { get; set; }
    }
    
    public enum CategoryType: byte
    {
        Weapon,
        Armor
    }
}
